contractcourt/channel_arbitrator.go

```diff
@@ -1131,11 +1139,102 @@ func (c *ChannelArbitrator) sweepAnchors(anchors []*lnwallet.AnchorResolution,
 		if err != nil {
 			return err
 		}
+
+		return nil
+	}
+
+	// Calculate the deadline. This is only valid for sweeping the local
+	// anchor.
+	deadline, err := c.findCommitmentDeadline(heightHint)
+	if err != nil {
+		return err
+	}
+
+	for _, anchor := range anchors {
+		// The first item is the anchor of the local commitment tx, we
+		// will use the deadline calcuated from the active HTLCs.
+		if err := sweepWithDeadline(anchor, deadline); err != nil {
+			return err
+		}
+
+		// For the rest anchors, use the default conf target.
+		deadline = uint32(anchorSweepConfTarget)
 	}
 
 	return nil
 }
 
+// findCommitmentDeadline finds the deadline for a commitment transaction by
+// extracting the minimum CLTV from its HTLCs. Currently only the local
+// commitment transaction is suitable to have a confirmation deadline. And it's
+// defined to be the smaller of,
+//   - the least CLTV from outgoing HTLCs, or,
+//   - the least CLTV from incoming HTLCs if the preimage is available.
+func (c *ChannelArbitrator) findCommitmentDeadline(
```

Comment:
This function can also be incorporated into `c.checkCommitChainActions` as we are checking all the HTLCs there anyway. However I think it's best to make its own function for clarity.

Reply:
I like this shift to house the logic in the channel arb itself as well, can allow us to unit test the component more discretely.